### PR TITLE
Implement List Section in the ASR Pass

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Fixes: #1177

### CODE REVIEW

File: integration_tests/test_list_10.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    i: i32
+    for i in range(50):
+        x.append(i)
+
+    y = x[15:35]
+    j: i32 = 0
+    for i in range(15, 35):
+        assert x[i] == y[j]
+        j += 1
+
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/test_list_10.py:24]: @czgdp1807, do you suggest any other tests?

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
Please rename,

1. `class ListASRPass` -> `class ListExprReplacer`
2. `class ListConcatVisitor` -> `class ListExprVisitor`
3. `pass_list_concat` -> `pass_list_expr`

The implementation and the tests look fine to me.

**Comment by Thirumalai-Shaktivel** [comment]
@czgdp1807, I applied your suggestions and provided support for list section step `x[::2]`. If everything looks good I will start polishing the commit

**Review by czgdp1807 — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the review!

**Comment by czgdp1807** [comment]
Please wait for https://github.com/lfortran/lfortran/pull/875 to be merged.

**Comment by Thirumalai-Shaktivel** [comment]
Yup, I will polish the commit and keep it.

**Pull request merged**